src/enum-thread.c

```diff
@@ -407,8 +407,33 @@ int enum_threads_init (enum_t *E) {
       E->threads[t].state[i].energy = 0.0;
 
   /* compute the tree size. */
-  for (unsigned int i = 0; i < E->G->n_order; i++)
+  for (unsigned int i = 0; i < E->G->n_order; i++) {
+    if (i >= 3)  {
+       /* get the d(i,i-3) edge weight. */
+       const unsigned int i0 = E->G->order[i - 3];
+       const unsigned int i1 = E->G->order[i - 2];
+       const unsigned int i2 = E->G->order[i - 1];
+       const unsigned int i3 = E->G->order[i];
+
+       /* get the residue indices. */
+       unsigned int r0 = E->P->atoms[i0].res_id;
+       unsigned int r1 = E->P->atoms[i1].res_id;
+       unsigned int r2 = E->P->atoms[i2].res_id;
+       unsigned int r3 = E->P->atoms[i3].res_id;
+       info("r0:%d, r1:%d, r2:%d, r3:%d",r0+1,r1+1,r2+1,r3+1);
```

Comment:
This is totally minor, but could you please include spaces between the arguments to `info()` here? i.e.:
```suggestion
       info("r0:%d, r1:%d, r2:%d, r3:%d", r0 + 1, r1 + 1, r2 + 1, r3 + 1);
```